src/data/ab-attrs/bonus-crit-ab-attr.ts

```diff
@@ -1,16 +1,25 @@
 import type { Pokemon } from "#app/field/pokemon";
-import type { BooleanHolder } from "#app/utils";
+import type { NumberHolder } from "#app/utils";
 import { AbAttrFlag } from "#enums/ab-attr-flag";
 import { AbAttr } from "./ab-attr";
 
+/**
+ * Ability attribute that provides bonus critical hit rate stages to the ability holder
+ * It is used by the ability Super Luck, which provides a one stage boost to critical hit rate.
+ * @extends AbAttr
+ */
 export class BonusCritAbAttr extends AbAttr {
-  constructor(showAbility: boolean = true, showAbilityInstant: boolean = false) {
+  /** Additional critical hit stages provided by the ability. */
+  private stages: number; 
```

Comment:
NIT: Does it ever change after being set in the constructor? If not add `readonly`